llvm/lib/Analysis/ConstantFolding.cpp

```diff
@@ -3624,6 +3628,10 @@ bool llvm::isMathLibCallNoop(const CallBase *Call,
       case LibFunc_sqrtf:
         return Op.isNaN() || Op.isZero() || !Op.isNegative();
 
+      case LibFunc_erf:
+      case LibFunc_erff:
+        return true;
```

Comment:
I'm unclear on if and when errno is written. The linux man page says this. The C standard seems to suggest it raises a range error, which depending on platform configuration may be errno and/or fp exception

Reply:
I don't think this is actually that safe, llvm isn't compiling with strictfp enabled

Reply:
Yes, it will probably work on most hosts in practice but it's not structurally correct

Reply:
You would have to enable strictfp support (which barely works in llvm), or avoid calling the host function in any of the edge cases that you are relying on the exception